docs/spec-files/general_messaging_constructs.md

```diff
@@ -24,19 +24,18 @@ Attachments are contained within a list in the `attachments` header.
 - `description`: [optional] A human-readable description of the content.
 - `filename`: A hint about the name that might be used if this attachment is persisted as a file. It is not required, and need not be unique. If this field is present and `mime-type` is not, the extension on the filename may be used to infer a MIME type.
-- `mime_type`: [optional] Describes the MIME type of the attached content.
-- `format`: [optional] Describes the format of the attachment if the `mime_type` is not sufficient.
-- `lastmod_time`: A hint about when the content in this attachment was last modified.
+- `media_type`: [optional] Describes the media (MIME) type of the attached content.
+- `format`: [optional] Describes the format of the attachment if the `media_type` is not sufficient.
+- `lastmod_time`: [optional] A hint about when the content in this attachment was last modified.
 - `byte_count`: [optional] mostly relevant when content is included by reference instead of by value. Lets the receiver guess how expensive it will be, in time, bandwidth, and storage, to fully fetch the attachment.
-- `data`: A JSON object that gives access to the actual content of the attachment. Contains the following subfields:
-  - `jws`: [optional] A JSON Web Signature over the content of the attachment.
+- `data`: A JSON object that gives access to the actual content of the attachment. Contains enough of the following subfields to allow access to the data:
+  - `jws`: [optional] A [JWS](https://tools.ietf.org/html/rfc7515) in [detached dontent mode](https://tools.ietf.org/html/rfc7515#appendix-F), where the `payload` field of the JWS maps to `base64` or to something fetchable via `links`. This allows attachments to be signed. The signature need not come from the author of the message.
   - `hash`: [optional] The hash of the content encoded in multi-hash format. Used as an integrity check for the attachment, and MUST be used if the data is referenced via the `links` data attribute. 
   - `links`: [optional] A list of zero or more locations at which the content may be fetched.
-  - `base64`: [optional]Base64-encoded data, when representing arbitrary content inline instead of via `links`.
-  - `jwe`: Already encrypted data in the form of a JWE.
```

Comment:
Would we need this for when we're sending a JWE JSON message?

Reply:
ie. forward protocol I believe uses this?

it is just json though.. which begs the question why have all of these keys at all. why not just have 3? `id`, `type`, `base64`

Reply:
I actually proposed just `base64`, but was outvoted. I still like that.

I *could* make one argument for `json`: it prevents an extra 33% bloat. If I put a JWE into `base64`, I have to base64 a bunch of stuff that's already base64'ed internally, so the bloat is worse. Having `json` lets me skip this.

But I can't really see *any* justification for both `json` and `jwe`. JWE is just JSON -- and we can already tell them apart using `media-type`.

Reply:
I too wonder what the need for the property to describe it inside data is for. Now that we have the mime-type I wonder if it's useful for us to just put all the data as the value in the "data" property and make the value polymorphic (e.g. could be a JSON array, string, number, object)

Reply:
@kdenhartog I'd expect pushback on having a value be of a variable type for those using inflexible serialization engines.